src/journey-maps/angular/services/map/map-journey-service.ts

```diff
@@ -42,14 +51,72 @@ export class SbbMapJourneyService {
       }
     }
 
-    this._mapRouteService.updateRoute(map, mapSelectionEventService, {
-      type: 'FeatureCollection',
-      features: routeFeatures,
-    });
+    if (isV1Style(map)) {
+      this._sourceRouteService.updateRoute(map, mapSelectionEventService, {
+        type: 'FeatureCollection',
+        features: routeFeatures,
+      });
+
+      this._transferSourceService.updateTransferV1(map, {
+        type: 'FeatureCollection',
+        features: transferFeatures,
+      });
+    } else {
+      this._sourceRouteService.updateRoute(map, mapSelectionEventService, {
+        type: 'FeatureCollection',
+        features: routeFeatures.concat(transferFeatures),
+      });
 
-    this._mapTransferService.updateTransfer(map, {
-      type: 'FeatureCollection',
-      features: transferFeatures,
+      if (journeyMetaInformation?.selectedLegId) {
```

Comment:
Does it work when switching from a selected leg to no selected leg?

Reply:
nope.